### PR TITLE
Delete comment

### PR DESCRIPTION
**Pull request by mohammedali101**
delete comment functionality added
total number of characters for comments added
making delete comment be reactive

### CODE REVIEW

**Comment by mohammedali101** [comment]
https://user-images.githubusercontent.com/91173297/201543655-ffd3e564-89be-463e-bd25-2ef9ac13b997.mov

Not included in video, but user is not allowed to exceed 256 characters in the comment body because of inputFormatters

image below shows user can only delete their comment
user can only delete their comment

**Comment by mohammedali101** [comment]
https://user-images.githubusercontent.com/91173297/201554305-a73b6ad5-963a-4b2b-bbea-39bc2ca8203e.mov

File: lib/util/apis.dart [hunk 1]
```diff
@@ -237,6 +237,34 @@ Future<Map<String, dynamic>> createComment(String commentBody, String postId, St
...
+      "commentId": commentId,
+    }),
+    headers: {
+      "Authorization": token,
+      "Content-Type": "application/json",
+    },
+  ).then(
+    (response) {
+      if (response.statusCode == 200) {
+        print("Success!");
```
Comment (tanvirahmed725) [inline lib/util/apis.dart:255]: Lets get rid of unnecessary print statements.

File: lib/pages/post_widgets/post_widget.dart [hunk 2]
```diff
@@ -64,11 +65,21 @@ class _PostWidget extends State<PostWidget> {
         deletePost(widget.post.postId, widget.token);
     responseMessage.then((response) {
       if (response['status'] == true) {
+        print("Post has successfully been deleted!");
```
Comment (tanvirahmed725) [inline lib/pages/post_widgets/post_widget.dart:68]: Let's get rid of unnecessary print statements.

File: lib/pages/post_page/post_page.dart [hunk 3]
```diff
@@ -48,6 +50,21 @@ class PostPageState extends State<PostPage> {
     Navigator.pushReplacement(context, MaterialPageRoute(builder: (context) => widget.route));
   }
 
+doDeleteComment(passIndex) {
+    final Future<Map<String, dynamic>> responseMessage = deleteComment(widget.postId, _comments[passIndex]['commentId'], widget.token);
+    responseMessage.then((response) {
+      if (response['status'] == true) {
+        print("Comment has successfully been deleted!");
```
Comment (tanvirahmed725) [inline lib/pages/post_page/post_page.dart:57]: Unnecessary print statement.

File: lib/pages/post_widgets/post_widget.dart [hunk 4]
```diff
@@ -64,11 +65,21 @@ class _PostWidget extends State<PostWidget> {
...
+          title: Text("Error!"),
+          content: Text('An error occurred while deleting the post, please try again.'),
+        );
+      }
     });
   }
 
+  delete() {
+    Navigator.pop(context);
+  }
```
Comment (tanvirahmed725) [inline lib/pages/post_widgets/post_widget.dart:81]: lets get rid of this function

**Review by tanvirahmed725 — CHANGES_REQUESTED** [review]

**Pull request merged**